### PR TITLE
feat(triton): add fused paged TQ4 decode attention kernel

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Decode attention on compressed TQ4 KV caches required decompressing to
HBM first, then running Flash Attention — three serial HBM operations
per token (read compressed, write FP16, re-read FP16). This fused kernel
reads compressed blocks directly from vLLM's page table, decompresses in
SRAM, and computes attention in a single pass, eliminating the HBM
round-trip.

- Add fused paged TQ4 decode kernel with tensor-of-pointers paged block access, in-tile decompression (nibble unpack, centroid gather, norm scale), and online softmax
- Add Python wrapper with pre/post rotation and correct `out` parameter contract (scratch buffer + copy-back)
- Add 16 tests: cache parity (>0.999), kernel correctness (>0.998), edge cases, masking, multi-seq batch, 36-layer composition, bf16, MHA
- Document HBM traffic reduction: 136 bytes/token (fused) vs 1,160 bytes/token (reference) = 8.5x

Test: `uv run pytest tests/test_fused_paged_tq4_attention.py -v`

fix(triton): resolve code review findings for fused paged TQ4 kernel

- Fix `out` parameter to write final post-rotated result into caller's buffer
- Shuffle physical block indices in test fixture to validate non-contiguous paged access
- Replace duplicate GQA test config with MHA (4Q/4KV) boundary case
- Add bf16 dtype parametrization to cache parity test

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [ ] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- Kernel correctness: tensor-of-pointers paged access pattern with non-contiguous physical blocks
- `out` parameter contract: scratch buffer approach ensures caller gets post-rotated result
- Online softmax fp32 discipline with exp2 fast path
- bf16 support through cache parity test only (kernel correctness test reference path not bf16-validated)

### Related
- Story 6.1: Consolidated kernel design document (prerequisite)
- Story 6.3: Backend integration and feature gating (next)
- Experiment 018: CUDA graph decode latency baseline (motivates fused approach)
- Experiment 019: HBM traffic analysis (included in this PR)

### CODE REVIEW

File: experiments/logs/experiment-019-fused-paged-hbm-traffic.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+The fused paged TQ4 decode kernel reads compressed blocks directly from
+vLLM's page table and decompresses in SRAM, eliminating the HBM
+round-trip for decompressed FP16 data.
+
+## Per-Token HBM Traffic
+
+| Path | Read (B) | Write (B) | Total (B) |
+|------|----------|-----------|-----------|
+| **Fused paged** | 136 | 0 | **136** |
+| Reference (decompress-all) | 136 | 512 | 1,160* |
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply experiments/logs/experiment-019-fused-paged-hbm-traffic.md]: False positive — the table uses standard single `|` separators and renders correctly in GitHub.

File: src/turboquant_vllm/triton/fused_paged_tq4_attention.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+decompressed cache -- HBM traffic drops from 1,160 to 136 bytes/token
+(8.5x reduction).
+
+The kernel operates entirely in **rotated space**.  The caller
+pre-rotates Q by ``Pi^T`` and post-rotates the output by ``Pi``.
+Decompression does NOT apply rotation (matching ``tq4_decompress.py``).
+
+Scope: FP16 Q decode path only (``USE_INT8_QK=False``).  INT8 path
+is Story 6.4.  Placeholder parameters are included for forward
+compatibility but compiled out by the constexpr switch.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply src/turboquant_vllm/triton/fused_paged_tq4_attention.py]: Valid — fixed in cleanup commit: docstring updated to 'FP16/BF16 Q decode path only'.

File: experiments/logs/experiment-019-fused-paged-hbm-traffic.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+### Fused path (single kernel, no HBM temp buffer)
+- K compressed read: 4 heads * 68 bytes = 272 bytes (indices + norms)
+- V compressed read: 4 heads * 68 bytes = 272 bytes
+- Total per token: 544 bytes across all heads
+- Per K or V per head: 68 bytes (64 nibble-packed + 4 fp32 norm)
+
+### Reference path (three serial HBM operations)
+1. **Read** compressed KV from HBM: 136 bytes/token/head-pair
+2. **Write** decompressed FP16 KV to HBM temp: 512 bytes/token
+3. **Read** decompressed FP16 KV for Flash Attention: 512 bytes/token
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply experiments/logs/experiment-019-fused-paged-hbm-traffic.md]: Valid — fixed in cleanup commit: clarified 'Total per token (all 4 KV head pairs): 544 bytes'.
